packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/text/TextLayoutManager.java

```diff
@@ -105,96 +104,9 @@ private static void buildSpannableFromFragment(
-      } else if (end >= start) {
-        boolean roleIsLink =
-            textAttributes.mRole != null
-                ? textAttributes.mRole == Role.LINK
-                : textAttributes.mAccessibilityRole == AccessibilityRole.LINK;
-        if (roleIsLink) {
-          ops.add(new SetSpanOperation(start, end, new ReactClickableSpan(reactTag)));
-        }
-        if (textAttributes.mIsColorSet) {
-          ops.add(
-              new SetSpanOperation(
-                  start, end, new ReactForegroundColorSpan(textAttributes.mColor)));
-        }
-        if (textAttributes.mIsBackgroundColorSet) {
-          ops.add(
-              new SetSpanOperation(
-                  start, end, new ReactBackgroundColorSpan(textAttributes.mBackgroundColor)));
-        }
-        if (!Float.isNaN(textAttributes.getLetterSpacing())) {
-          ops.add(
-              new SetSpanOperation(
-                  start, end, new CustomLetterSpacingSpan(textAttributes.getLetterSpacing())));
-        }
-        ops.add(
-            new SetSpanOperation(start, end, new ReactAbsoluteSizeSpan(textAttributes.mFontSize)));
-        if (textAttributes.mFontStyle != UNSET
-            || textAttributes.mFontWeight != UNSET
-            || textAttributes.mFontFamily != null) {
-          ops.add(
-              new SetSpanOperation(
-                  start,
-                  end,
-                  new CustomStyleSpan(
-                      textAttributes.mFontStyle,
-                      textAttributes.mFontWeight,
-                      textAttributes.mFontFeatureSettings,
-                      textAttributes.mFontFamily,
-                      context.getAssets())));
-        }
-        if (textAttributes.mIsUnderlineTextDecorationSet) {
-          ops.add(new SetSpanOperation(start, end, new ReactUnderlineSpan()));
-        }
-        if (textAttributes.mIsLineThroughTextDecorationSet) {
-          ops.add(new SetSpanOperation(start, end, new ReactStrikethroughSpan()));
-        }
-        if ((textAttributes.mTextShadowOffsetDx != 0
-                || textAttributes.mTextShadowOffsetDy != 0
-                || textAttributes.mTextShadowRadius != 0)
-            && Color.alpha(textAttributes.mTextShadowColor) != 0) {
-          ops.add(
-              new SetSpanOperation(
-                  start,
-                  end,
-                  new ShadowStyleSpan(
-                      textAttributes.mTextShadowOffsetDx,
-                      textAttributes.mTextShadowOffsetDy,
-                      textAttributes.mTextShadowRadius,
-                      textAttributes.mTextShadowColor)));
-        }
-        if (!Float.isNaN(textAttributes.getEffectiveLineHeight())) {
-          ops.add(
-              new SetSpanOperation(
-                  start, end, new CustomLineHeightSpan(textAttributes.getEffectiveLineHeight())));
-        }
+    final var textFragmentList = new BridgeTextFragmentList(fragments);
```

Comment:
Let's say you create a feature flag called enableSpannableCreationUnification then here's one of the places where you have to gate your new code:

if (enableSpannableCreationUnification) {
   textFragmentList = new BridgeTextFragmentList(fragments);
   ...
} else {
  for (int i = 0, length = fragments.size(); i < length; i++) {
  ....
  }
}
```

I know this make this code a bit dirty, but it would be important to understand the impact of this PR